packages/nearfi/src/lib/injected-nearfi.ts

```diff
@@ -0,0 +1,141 @@
+  helperUrl: string;
+  index: number;
+  name: string;
+  network: string;
+  networkId: string;
+  nodeUrl: string;
+  walletUrl: string;
+  wrapNearContract: string;
+}
+
+export interface GetRpcResponse {
+  method: "getRpc";
+  notificationId: number;
+  rpc: RpcInfo;
+  type: "nearfi-wallet-result";
+}
+
+export interface RequestSignInParams {
+  contractId: string;
+  methodNames?: Array<string>;
+  amount?: string; // in yoctoⓃ
+}
+
+export interface RpcChangedResponse {
+  explorerUrl: string;
+  helperUrl: string;
+  index: number;
+  name: string;
+  network: string;
+  networkId: string;
+  nodeUrl: string;
+  walletUrl: string;
+  wrapNearContract: string;
+}
+
+export interface SendMoneyParams {
+  receiverId: string;
+  amount: string;
+}
+
+export interface SendMoneyResponse {
+  transactionHash: string;
+  error?: string;
+}
+
+export interface Action {
+  methodName: string;
+  args: object;
+  gas: string;
+  deposit: string;
+}
+
+export interface SignAndSendTransactionParams {
+  receiverId: string;
+  actions: Array<Action>;
+}
+
+// Seems to reuse signAndSendTransactions internally, hence the wrong method name and list of responses.
+export interface SignAndSendTransactionResponse {
+  actionType: "DAPP/DAPP_POPUP_RESPONSE";
+  method: "signAndSendTransactions";
+  notificationId: number;
+  error?: string;
+  response?: Array<providers.FinalExecutionOutcome>;
+  type: "nearfi-wallet-extensionResult";
+}
+
+export interface SignAndSendTransactionsResponse {
+  actionType: "DAPP/DAPP_POPUP_RESPONSE";
+  method: "signAndSendTransactions";
+  notificationId: number;
+  error?: string;
+  response?: Array<providers.FinalExecutionOutcome>;
+  type: "nearfi-wallet-extensionResult";
+}
+
+export interface Transaction {
+  receiverId: string;
+  actions: Array<Action>;
+}
+
+export interface RequestSignTransactionsParams {
+  transactions: Array<Transaction>;
+}
+
+export interface NearFiEvents {
+  signIn: () => void;
+  signOut: () => void;
+  accountChanged: (changedAccountId: string) => void;
+  rpcChanged: (response: RpcChangedResponse) => void;
+}
+
+export interface InjectedNearFi {
+  isNearFi: boolean;
+  getAccountId: () => string | null;
+  getRpc: () => Promise<GetRpcResponse>;
+  requestSignIn: (
+    params: RequestSignInParams
+  ) => Promise<RequestSignInResponse>;
+  signOut: () => Promise<SignOutResponse>;
+  isSignedIn: () => boolean;
+  removeEventListener: (event: string) => void;
+  on: <Event extends keyof NearFiEvents>(
+    event: Event,
+    callback: NearFiEvents[Event]
+  ) => void;
+  // TODO: Determine return type.
+  sendMoney: (params: SendMoneyParams) => Promise<SendMoneyResponse>;
+  signAndSendTransaction: (
+    params: SignAndSendTransactionParams
+  ) => Promise<SignAndSendTransactionResponse>;
+  requestSignTransactions: (
+    params: RequestSignTransactionsParams
+  ) => Promise<SignAndSendTransactionsResponse>;
+  log: (...msg: any) => void;
+  resolveSignInState: () => Promise<any>;
```

Comment:
If you do `yarn lint` it will fail. You can't use `any`. Replace with something like `unknown` and try again to run `yarn lint`.

Reply:
done sir @amirsaran3